File: Cpp/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        ListNode* node = head;
        int result = 0;
        while(node){
            result = result * 2 + node->val;
            node = node->next;
        }
        
        return result;
    }
};

File: Kotlin/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun getDecimalValue(head: ListNode?): Int {
        var node = head
        var result = 0
        
        while(node != null){
            result = result * 2 + node.`val`
            node = node.next
        }
        
        return result
    }
}